spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/LoadBalancerWebClientHttpServiceGroupConfigurer.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.boot.autoconfigure.http.client.reactive.service.ReactiveHttpClientServiceProperties;
+import org.springframework.boot.autoconfigure.http.client.service.HttpClientServiceProperties;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.loadbalancer.LoadBalancerProperties;
+import org.springframework.cloud.client.loadbalancer.LoadBalancerUriTools;
+import org.springframework.util.function.SingletonSupplier;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.support.WebClientHttpServiceGroupConfigurer;
+
+import static org.springframework.cloud.client.loadbalancer.LoadBalancerUriTools.constructInterfaceClientsBaseUrl;
+import static org.springframework.cloud.client.loadbalancer.LoadBalancerUriTools.isServiceIdUrl;
+
+/**
+ * Load-balancer-specific {@link WebClientHttpServiceGroupConfigurer} implementation. If
+ * the group {@code baseUrl} is {@code null}, sets up a {@code baseUrl} with LoadBalancer
+ * {@code serviceId} -resolved from Interface Client {@code groupName} set as
+ * {@code host}. If the group {@code baseUrl} is {@code null} or
+ * {@link LoadBalancerUriTools#isServiceIdUrl(String, String)}, a
+ * {@link DeferringLoadBalancerExchangeFilterFunction} instance picked from application
+ * context is added to the group's {@link WebClient.Builder} if available, allowing for
+ * the requests to be load-balanced.
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 5.0.0
+ * @see WebClient.Builder
+ * @see HttpClientServiceProperties
+ */
+public class LoadBalancerWebClientHttpServiceGroupConfigurer implements WebClientHttpServiceGroupConfigurer {
+
+	// Make sure Boot's configurers run before
+	private static final int ORDER = 10;
+
+	private final ReactiveLoadBalancer.Factory<ServiceInstance> loadBalancerClientFactory;
+
+	private final SingletonSupplier<DeferringLoadBalancerExchangeFilterFunction<LoadBalancedExchangeFilterFunction>> loadBalancerFilterFunctionSupplier;
+
+	private final ReactiveHttpClientServiceProperties clientServiceProperties;
+
+	public LoadBalancerWebClientHttpServiceGroupConfigurer(
+			ObjectProvider<DeferringLoadBalancerExchangeFilterFunction<LoadBalancedExchangeFilterFunction>> exchangeFilterFunctionProvider,
+			ReactiveHttpClientServiceProperties clientServiceProperties,
+			ReactiveLoadBalancer.Factory<ServiceInstance> loadBalancerClientFactory) {
+		this.loadBalancerFilterFunctionSupplier = SingletonSupplier
+			.ofNullable(exchangeFilterFunctionProvider.getIfAvailable());
+		this.clientServiceProperties = clientServiceProperties;
+		this.loadBalancerClientFactory = loadBalancerClientFactory;
+	}
+
+	@Override
+	public void configureGroups(Groups<WebClient.Builder> groups) {
+		DeferringLoadBalancerExchangeFilterFunction<LoadBalancedExchangeFilterFunction> loadBalancerFilterFunction = loadBalancerFilterFunctionSupplier
```

Comment:
Same as above, when would this not be available?

Reply:
TBH, mainly if the user has manually played with the configuration and botched it.